sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -1503,10 +1506,32 @@ def unstack(self, *args, **kwargs):
 
 @populate_not_implemented(pd.core.groupby.generic.DataFrameGroupBy)
 class DeferredGroupBy(frame_base.DeferredFrame):
-  def __init__(self, expr, kwargs):
+  def __init__(self, expr, kwargs,
+               ungrouped: DeferredDataFrameOrSeries, projection=None):
     super(DeferredGroupBy, self).__init__(expr)
+    # This object represents the result of:
+    # ungrouped.groupby(level=list(range(ungrouped.index.nlevels),
+    #                   **kwargs)[projection]
+    self._ungrouped = ungrouped
+    self._projection = projection
     self._kwargs = kwargs
 
+  def __getattr__(self, name):
+    return DeferredGroupBy(expressions.ComputedExpression(
+        'groupby_project',
+        lambda gb: getattr(gb, name), [self._expr],
+        requires_partition_by=partitionings.Nothing(),
+        preserves_partition_by=partitionings.Singleton()),
+                           self._kwargs, self._ungrouped, name)
```

Comment:
This indentation seems odd. Perhaps add a newline after the DeferredGroupBy above?

Reply:
Good catch, done.